api/app/Models/Skill.php

```diff
@@ -108,13 +115,18 @@ public function getExperiencesAttribute()
         return $collection;
     }
 
-    public function poolsEssentialSkills() : BelongsToMany
+    public function poolsEssentialSkills(): BelongsToMany
     {
         return $this->belongsToMany(Pool::class, 'pools_essential_skills');
     }
 
-    public function poolsNonessentialSkills() : BelongsToMany
+    public function poolsNonessentialSkills(): BelongsToMany
     {
         return $this->belongsToMany(Pool::class, 'pools_essential_skills');
     }
+
+    public function getDetailsAttribute()
+    {
+        return $this->experience_skill_pivot->details;
+    }
```

Comment:
this does not appear to work and errors 

`
Object { file: "/var/www/html/api/app/Models/Skill.php", line: 130, call: "Illuminate\\Foundation\\Bootstrap\\HandleExceptions::handleError(8, 'Trying to get property \\'details\\' of non-object', '/var/www/html/api/app/Models/Skill.php', 130, array(0))" }`

Reply:
Hmm, do you have steps to reproduce? I took a chance and made a change but I'm not certain if it fixed this.

[check for skill pivot before accessing details](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3938/commits/e7dffe74ab6217ac032292a2c931217756d4201d)

Reply:
attempting to publish a pool from `/admin/pools/:id/edit` 

experience_skill_pivot appears to be null

![image](https://user-images.githubusercontent.com/40485260/189982925-4d680877-87ae-4824-b263-804f20b17c0f.png)

![image](https://user-images.githubusercontent.com/40485260/189983197-f879201b-ef29-4a88-a4ed-55267f2d56aa.png)

Reply:
the latest changes fix the error
one question, what is the role of the added accessor? I don't spot any instance where getDetailsAttribute is called

Reply:
It is used in the `SkillResource`

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e7dffe74ab6217ac032292a2c931217756d4201d/api/app/Http/Resources/SkillResource.php#L24-L26

Reply:
ah, I  thought it was going to be explicitly declared as ->getDetails() somewhere
thanks